src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.cpp

```diff
@@ -113,22 +114,12 @@ void jit_emitter::emitter_preamble(const std::vector<size_t>& in_idxs,
         aux_gpr_idxs.erase(aux_gpr_idxs.end() - 1);
     }
 
-    for (size_t i = 0; i < preserved_gpr_idxs.size(); ++i) {
-        h->str(Xbyak_aarch64::XReg(preserved_gpr_idxs[i]), pre_ptr(h->sp, -16));
-    }
-
```

Comment:
Hi Edward! Thanks for the comments! There are two reasons.

Firstly, when I was trying to understand how aux_gpr_idxs and preserved_gpr_idxs were designed in the first place, I referred to the x64 implementation in https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.cpp#L62-L154.
If I understand it right, aux_gpr_idxs is provided by the caller, the caller assures these registers are exclusively used otherwise the caller should save the registers beforehand. And judging from https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.cpp#L94-L108, preserved_gpr_idxs will be used only when the number of the caller provided aux_gpr_idxs are less than aux_vecs_count(), then the callee will push the free registers to aux_gpr_idxs to meet the aux_vecs_count(), and also push them to preserved_gpr_idxs to save them. I think for most cases, aux_gpr_idxs can provide enough registers. At least for the cases I have met, the preserved_gpr_idxs has not been used. Still preserved_gpr_idxs is valuable as an insurance.
However, when it comes to the aarch64 implementation, preserved_gpr_idxs will always save and retore the registers provided by aux_gpr_idxs again. It seems not necessary.

Secondly, the more important reason is that h->str and h->ldr instructions for saving and restoring registers are quite costly, and they are frequently called in every emitter. The performance impact is significant. For the models I tested in this ticket, if we always save and retore preserved_gpr_idxs, the time cost will be 3~6 times over avoiding that.

So I think the current implementation works and is necessary. Now that we've checked the size of aux_gpr_idxs in https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.cpp#L93 , we do nothing with preserved_gpr_idxs.

Further discussions are always welcomed. Thanks Edward!

Reply:
We definitely need to store auxiliary general-propose and vector registers. Lets discuss offline.

Reply:
Hi Edward! I've double checked all the places used `preserved_gpr_idxs`. As `preserved_gpr_idxs` in `jit_uni_eltwise_injector_f32` is only a namesake. `preserved_gpr_idxs` is actually used nowhere else. Please correct me, if I'm wrong. So I think `preserved_gpr_idxs` in `jit_emitter` should be used as described above. And from the perspective of performance, the revision here is necessary (the usage of `preserved_gpr_idxs` in `jit_emitter`  on master has performance issue for Snippets).

Another alternative approach is to implement `preserved_gpr_idxs` the same way as x64, which doesn't have performance issue.

Please elaborate a little bit, if you have concerns. Thanks in advance!

@dmitry-gorokhov @a-sidorova Please also feel free to discuss. Thanks!

Reply:
discussed, thanks